tools/limiter/limiter.go

```diff
@@ -0,0 +1,85 @@
+package limiter
+
+import (
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/berachain/offchain-sdk/tools/store"
+)
+
+type Limiter struct {
+	Store  store.Store
+	Rate   int
+	Period time.Duration
+}
+
+type Config struct {
+	Enabled          bool
+	TTL              time.Duration
+	Rate             int
+	Kind             string
+	RedisAddr        string
+	RedisClusterMode bool
+	ProxyCount       int
+}
+
+func New(config Config) *Limiter {
+	var lstore store.Store
+	switch config.Kind {
+	case "redis":
+		lstore = store.NewRedisStore(config.TTL, config.RedisAddr, config.RedisClusterMode)
```

Comment:
can you have it inferred so the `Kind` variable isn't necessary (enforces specific strings to be used). Just check if `RedisAddr != ""` then enable Redis, if not use mem

Reply:
sure